assets/dns/service.yaml

```diff
@@ -1,5 +1,8 @@
 kind: Service
 apiVersion: v1
+metadata:
+  annotations:
+    service.beta.openshift.io/serving-cert-secret-name: metrics-tls
```

Comment:
It seems strange that the name of the service associated with the `dns` resource is set at runtime, but the name of the associated secret is static.  Granted, in practice, the `dns` resource should be a singleton (the operator processes the `dnses.operator.openshift.io/default` object and ignores any other `dnses.operator.openshift.io` objects), and the singleton's name is always going to be the same (`default`), but as long as the operator is deriving the service's name from the dns's name, it seems like the secret's name should likewise be derived from the dns's name.  What do you think?